From 2f9424c78cf0edd8409b76fbef07667a8940ba82 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Sat, 19 Jul 2014 02:56:39 +1000
Subject: [PATCH 1/3] metadata: return both movie and TV results when in
 non-automatic mode

This allows not having to fiddle with season or episode #. TV or movie entries are presented first depending on the guessed type.
We only do so if multiple results were already found
---
 mythtv/libs/libmythmetadata/metadatadownload.cpp |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/mythtv/libs/libmythmetadata/metadatadownload.cpp b/mythtv/libs/libmythmetadata/metadatadownload.cpp
index 0f3a702..14bf08a 100644
--- a/mythtv/libs/libmythmetadata/metadatadownload.cpp
+++ b/mythtv/libs/libmythmetadata/metadatadownload.cpp
@@ -106,15 +106,17 @@ void MetadataDownload::run()
                 list = handleVideoUndetermined(lookup);
             }
 
-            if (list.isEmpty() &&
+            if ((list.isEmpty() ||
+                 (list.size() > 1 && !lookup->GetAutomatic())) &&
                 lookup->GetSubtype() == kProbableTelevision)
             {
-                list = handleMovie(lookup);
+                list.append(handleMovie(lookup));
             }
-            else if (list.isEmpty() &&
+            else if ((list.isEmpty() ||
+                      (list.size() > 1 && !lookup->GetAutomatic())) &&
                      lookup->GetSubtype() == kProbableMovie)
             {
-                list = handleTelevision(lookup);
+                list.append(handleTelevision(lookup));
             }
         }
         else if (lookup->GetType() == kMetadataGame)
-- 
1.7.10.2

